specification/cognitiveservices/data-plane/QnAMaker/stable/v4.0/QnAMaker.json

```diff
@@ -0,0 +1,1275 @@
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful query": {
+            "$ref": "./examples/SuccessfulDelKb.json"
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "Knowledgebases"
+        ],
+        "summary": "Publishes all changes in test index of a knowledgebase to its prod index.",
+        "operationId": "Publish Knowledgebase",
+        "parameters": [
+          {
+            "$ref": "#/parameters/KbId"
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "204": {
+            "description": "HTTP 204 No content."
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful query": {
+            "$ref": "./examples/SuccessfulPubKb.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Knowledgebases"
+        ],
+        "summary": "Replace knowledgebase contents.",
+        "operationId": "Replace Knowledgebase",
+        "parameters": [
+          {
+            "$ref": "#/parameters/KbId"
+          },
+          {
+            "$ref": "#/parameters/ReplaceKb"
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "204": {
+            "description": "HTTP 204 No content."
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Successful query": {
+            "$ref": "./examples/SuccessfulRepKb.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Knowledgebases"
+        ],
+        "summary": "Asynchronous operation to modify a knowledgebase.",
+        "operationId": "Update Knowledgebase",
+        "parameters": [
+          {
+            "$ref": "#/parameters/KbId"
+          },
+          {
+            "$ref": "#/parameters/UpdateKb"
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "202": {
+            "description": "Details of the asynchronous operation.",
+            "schema": {
+              "$ref": "#/definitions/Operation"
+            },
+            "x-nullable": true,
+            "headers": {
+              "Location": {
```

Comment:
since this is data plane, not sure if you're doing a long running operation this similar to ARM, if so, you can check out https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-long-running-operation

Reply:
good idea I looked at it but our api is returning relative path... Is this required ? we would need to make changes to our api to return absolute path.

Reply:
don't worry then, I just thought I would share in case it was helpful